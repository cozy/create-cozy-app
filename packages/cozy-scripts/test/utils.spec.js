/* eslint-env jest */

const fs = require('fs')
const paths = require('../utils/paths')
const CTS = require('../utils/constants.js')

describe('Paths provider helper', () => {
  beforeEach(() => {
    delete process.env[CTS.ENTRY_EXT]
    delete process.env[CTS.SRC_DIR]
    delete process.env[CTS.TEST_DIR]
    delete process.env[CTS.BUILD_DIR]
    delete process.env[CTS.MANIFEST]
  })

  // the app path will always be the same here
  const replaceAppPath = path =>
    path.replace(fs.realpathSync(process.cwd()), '/Custom/Path/To/App')

  it('should provide all default paths', () => {
    const pathsList = []
    for (let getter in paths) {
      if (paths.hasOwnProperty(getter)) {
        pathsList.push(replaceAppPath(paths[getter]()))
      }
    }
    expect(pathsList).toMatchSnapshot()
  })

  //environment variables handling for options
  ;[
    [CTS.ENTRY_EXT, '.vue'],
    [CTS.SRC_DIR, 'subfolder/src/myapp'],
    [CTS.TEST_DIR, 'subfolder/test/mytests'],
    [CTS.BUILD_DIR, 'subfolder/build/myapp'],
    [CTS.MANIFEST, 'subfolder/src/myapp/manifest.webapp']
  ].map(params => {
    it(`should provide all paths with custom ${params[0]}`, () => {
      process.env[params[0]] = params[1]
      const pathsList = []
      for (let getter in paths) {
        if (paths.hasOwnProperty(getter)) {
          pathsList.push(replaceAppPath(paths[getter]()))
        }
      }
      expect(pathsList).toMatchSnapshot()
    })
  })
})
