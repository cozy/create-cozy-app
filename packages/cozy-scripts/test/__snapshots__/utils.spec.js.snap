// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paths provider helper should provide all default paths 1`] = `
Array [
  "/Custom/Path/To/App",
  "/Custom/Path/To/App/build",
  "/Custom/Path/To/App/build/services",
  "/Custom/Path/To/App/build/assets/json",
  "/Custom/Path/To/App/manifest.webapp",
  "/Custom/Path/To/App/README.md",
  "/Custom/Path/To/App/LICENSE",
  "/Custom/Path/To/App/src",
  "/Custom/Path/To/App/test",
  "/Custom/Path/To/App/src/locales",
  "/Custom/Path/To/App/src/targets/vendor/assets",
  "/Custom/Path/To/App/node_modules",
  "/Custom/Path/To/App/src/targets/browser/index.ejs",
  "/Custom/Path/To/App/src/targets/browser/index.js",
  "/Custom/Path/To/App/src/targets/intents/index.ejs",
  "/Custom/Path/To/App/src/targets/intents/index.js",
  "/Custom/Path/To/App/src/targets/public/index.ejs",
  "/Custom/Path/To/App/src/targets/public/index.js",
  "/Custom/Path/To/App/src/targets/services",
  "/Custom/Path/To/App/src/targets/mobile/index.ejs",
  "/Custom/Path/To/App/src/targets/mobile/index.js",
  "/Custom/Path/To/App/src/targets/mobile/www",
  "/Custom/Path/To/App/node_modules/cozy-bar/dist/cozy-bar.js",
  "/Custom/Path/To/App/node_modules/cozy-bar/dist/cozy-bar.css",
  "/Custom/Path/To/App/node_modules/cozy-client-js/dist/cozy-client.js",
  "/Custom/Path/To/App/node_modules/cozy-ui/stylus",
  "/Custom/Path/To/App/packages/cozy-scripts/stack/disableCSP.yaml",
  "/Custom/Path/To/App/packages/cozy-scripts/node_modules/.bin/eslint",
  "/Custom/Path/To/App/packages/cozy-scripts/utils/reactExposer.js",
  "/Custom/Path/To/App/packages/cozy-scripts/stack/quitStack.sh",
]
`;

exports[`Paths provider helper should provide all paths with custom COZY_SCRIPTS_APP_BUILD_DIR 1`] = `
Array [
  "/Custom/Path/To/App",
  "/Custom/Path/To/App/subfolder/build/myapp",
  "/Custom/Path/To/App/subfolder/build/myapp/services",
  "/Custom/Path/To/App/subfolder/build/myapp/assets/json",
  "/Custom/Path/To/App/manifest.webapp",
  "/Custom/Path/To/App/README.md",
  "/Custom/Path/To/App/LICENSE",
  "/Custom/Path/To/App/src",
  "/Custom/Path/To/App/test",
  "/Custom/Path/To/App/src/locales",
  "/Custom/Path/To/App/src/targets/vendor/assets",
  "/Custom/Path/To/App/node_modules",
  "/Custom/Path/To/App/src/targets/browser/index.ejs",
  "/Custom/Path/To/App/src/targets/browser/index.js",
  "/Custom/Path/To/App/src/targets/intents/index.ejs",
  "/Custom/Path/To/App/src/targets/intents/index.js",
  "/Custom/Path/To/App/src/targets/public/index.ejs",
  "/Custom/Path/To/App/src/targets/public/index.js",
  "/Custom/Path/To/App/src/targets/services",
  "/Custom/Path/To/App/src/targets/mobile/index.ejs",
  "/Custom/Path/To/App/src/targets/mobile/index.js",
  "/Custom/Path/To/App/src/targets/mobile/www",
  "/Custom/Path/To/App/node_modules/cozy-bar/dist/cozy-bar.js",
  "/Custom/Path/To/App/node_modules/cozy-bar/dist/cozy-bar.css",
  "/Custom/Path/To/App/node_modules/cozy-client-js/dist/cozy-client.js",
  "/Custom/Path/To/App/node_modules/cozy-ui/stylus",
  "/Custom/Path/To/App/packages/cozy-scripts/stack/disableCSP.yaml",
  "/Custom/Path/To/App/packages/cozy-scripts/node_modules/.bin/eslint",
  "/Custom/Path/To/App/packages/cozy-scripts/utils/reactExposer.js",
  "/Custom/Path/To/App/packages/cozy-scripts/stack/quitStack.sh",
]
`;

exports[`Paths provider helper should provide all paths with custom COZY_SCRIPTS_APP_MANIFEST 1`] = `
Array [
  "/Custom/Path/To/App",
  "/Custom/Path/To/App/build",
  "/Custom/Path/To/App/build/services",
  "/Custom/Path/To/App/build/assets/json",
  "/Custom/Path/To/App/subfolder/src/myapp/manifest.webapp",
  "/Custom/Path/To/App/README.md",
  "/Custom/Path/To/App/LICENSE",
  "/Custom/Path/To/App/src",
  "/Custom/Path/To/App/test",
  "/Custom/Path/To/App/src/locales",
  "/Custom/Path/To/App/src/targets/vendor/assets",
  "/Custom/Path/To/App/node_modules",
  "/Custom/Path/To/App/src/targets/browser/index.ejs",
  "/Custom/Path/To/App/src/targets/browser/index.js",
  "/Custom/Path/To/App/src/targets/intents/index.ejs",
  "/Custom/Path/To/App/src/targets/intents/index.js",
  "/Custom/Path/To/App/src/targets/public/index.ejs",
  "/Custom/Path/To/App/src/targets/public/index.js",
  "/Custom/Path/To/App/src/targets/services",
  "/Custom/Path/To/App/src/targets/mobile/index.ejs",
  "/Custom/Path/To/App/src/targets/mobile/index.js",
  "/Custom/Path/To/App/src/targets/mobile/www",
  "/Custom/Path/To/App/node_modules/cozy-bar/dist/cozy-bar.js",
  "/Custom/Path/To/App/node_modules/cozy-bar/dist/cozy-bar.css",
  "/Custom/Path/To/App/node_modules/cozy-client-js/dist/cozy-client.js",
  "/Custom/Path/To/App/node_modules/cozy-ui/stylus",
  "/Custom/Path/To/App/packages/cozy-scripts/stack/disableCSP.yaml",
  "/Custom/Path/To/App/packages/cozy-scripts/node_modules/.bin/eslint",
  "/Custom/Path/To/App/packages/cozy-scripts/utils/reactExposer.js",
  "/Custom/Path/To/App/packages/cozy-scripts/stack/quitStack.sh",
]
`;

exports[`Paths provider helper should provide all paths with custom COZY_SCRIPTS_APP_SRC_DIR 1`] = `
Array [
  "/Custom/Path/To/App",
  "/Custom/Path/To/App/build",
  "/Custom/Path/To/App/build/services",
  "/Custom/Path/To/App/build/assets/json",
  "/Custom/Path/To/App/manifest.webapp",
  "/Custom/Path/To/App/README.md",
  "/Custom/Path/To/App/LICENSE",
  "/Custom/Path/To/App/subfolder/src/myapp",
  "/Custom/Path/To/App/test",
  "/Custom/Path/To/App/subfolder/src/myapp/locales",
  "/Custom/Path/To/App/subfolder/src/myapp/targets/vendor/assets",
  "/Custom/Path/To/App/node_modules",
  "/Custom/Path/To/App/subfolder/src/myapp/targets/browser/index.ejs",
  "/Custom/Path/To/App/subfolder/src/myapp/targets/browser/index.js",
  "/Custom/Path/To/App/subfolder/src/myapp/targets/intents/index.ejs",
  "/Custom/Path/To/App/subfolder/src/myapp/targets/intents/index.js",
  "/Custom/Path/To/App/subfolder/src/myapp/targets/public/index.ejs",
  "/Custom/Path/To/App/subfolder/src/myapp/targets/public/index.js",
  "/Custom/Path/To/App/subfolder/src/myapp/targets/services",
  "/Custom/Path/To/App/subfolder/src/myapp/targets/mobile/index.ejs",
  "/Custom/Path/To/App/subfolder/src/myapp/targets/mobile/index.js",
  "/Custom/Path/To/App/subfolder/src/myapp/targets/mobile/www",
  "/Custom/Path/To/App/node_modules/cozy-bar/dist/cozy-bar.js",
  "/Custom/Path/To/App/node_modules/cozy-bar/dist/cozy-bar.css",
  "/Custom/Path/To/App/node_modules/cozy-client-js/dist/cozy-client.js",
  "/Custom/Path/To/App/node_modules/cozy-ui/stylus",
  "/Custom/Path/To/App/packages/cozy-scripts/stack/disableCSP.yaml",
  "/Custom/Path/To/App/packages/cozy-scripts/node_modules/.bin/eslint",
  "/Custom/Path/To/App/packages/cozy-scripts/utils/reactExposer.js",
  "/Custom/Path/To/App/packages/cozy-scripts/stack/quitStack.sh",
]
`;

exports[`Paths provider helper should provide all paths with custom COZY_SCRIPTS_APP_TEST_DIR 1`] = `
Array [
  "/Custom/Path/To/App",
  "/Custom/Path/To/App/build",
  "/Custom/Path/To/App/build/services",
  "/Custom/Path/To/App/build/assets/json",
  "/Custom/Path/To/App/manifest.webapp",
  "/Custom/Path/To/App/README.md",
  "/Custom/Path/To/App/LICENSE",
  "/Custom/Path/To/App/src",
  "/Custom/Path/To/App/subfolder/test/mytests",
  "/Custom/Path/To/App/src/locales",
  "/Custom/Path/To/App/src/targets/vendor/assets",
  "/Custom/Path/To/App/node_modules",
  "/Custom/Path/To/App/src/targets/browser/index.ejs",
  "/Custom/Path/To/App/src/targets/browser/index.js",
  "/Custom/Path/To/App/src/targets/intents/index.ejs",
  "/Custom/Path/To/App/src/targets/intents/index.js",
  "/Custom/Path/To/App/src/targets/public/index.ejs",
  "/Custom/Path/To/App/src/targets/public/index.js",
  "/Custom/Path/To/App/src/targets/services",
  "/Custom/Path/To/App/src/targets/mobile/index.ejs",
  "/Custom/Path/To/App/src/targets/mobile/index.js",
  "/Custom/Path/To/App/src/targets/mobile/www",
  "/Custom/Path/To/App/node_modules/cozy-bar/dist/cozy-bar.js",
  "/Custom/Path/To/App/node_modules/cozy-bar/dist/cozy-bar.css",
  "/Custom/Path/To/App/node_modules/cozy-client-js/dist/cozy-client.js",
  "/Custom/Path/To/App/node_modules/cozy-ui/stylus",
  "/Custom/Path/To/App/packages/cozy-scripts/stack/disableCSP.yaml",
  "/Custom/Path/To/App/packages/cozy-scripts/node_modules/.bin/eslint",
  "/Custom/Path/To/App/packages/cozy-scripts/utils/reactExposer.js",
  "/Custom/Path/To/App/packages/cozy-scripts/stack/quitStack.sh",
]
`;

exports[`Paths provider helper should provide all paths with custom COZY_SCRIPTS_ENTRY_EXT 1`] = `
Array [
  "/Custom/Path/To/App",
  "/Custom/Path/To/App/build",
  "/Custom/Path/To/App/build/services",
  "/Custom/Path/To/App/build/assets/json",
  "/Custom/Path/To/App/manifest.webapp",
  "/Custom/Path/To/App/README.md",
  "/Custom/Path/To/App/LICENSE",
  "/Custom/Path/To/App/src",
  "/Custom/Path/To/App/test",
  "/Custom/Path/To/App/src/locales",
  "/Custom/Path/To/App/src/targets/vendor/assets",
  "/Custom/Path/To/App/node_modules",
  "/Custom/Path/To/App/src/targets/browser/index.ejs",
  "/Custom/Path/To/App/src/targets/browser/index.vue",
  "/Custom/Path/To/App/src/targets/intents/index.ejs",
  "/Custom/Path/To/App/src/targets/intents/index.vue",
  "/Custom/Path/To/App/src/targets/public/index.ejs",
  "/Custom/Path/To/App/src/targets/public/index.vue",
  "/Custom/Path/To/App/src/targets/services",
  "/Custom/Path/To/App/src/targets/mobile/index.ejs",
  "/Custom/Path/To/App/src/targets/mobile/index.vue",
  "/Custom/Path/To/App/src/targets/mobile/www",
  "/Custom/Path/To/App/node_modules/cozy-bar/dist/cozy-bar.js",
  "/Custom/Path/To/App/node_modules/cozy-bar/dist/cozy-bar.css",
  "/Custom/Path/To/App/node_modules/cozy-client-js/dist/cozy-client.js",
  "/Custom/Path/To/App/node_modules/cozy-ui/stylus",
  "/Custom/Path/To/App/packages/cozy-scripts/stack/disableCSP.yaml",
  "/Custom/Path/To/App/packages/cozy-scripts/node_modules/.bin/eslint",
  "/Custom/Path/To/App/packages/cozy-scripts/utils/reactExposer.js",
  "/Custom/Path/To/App/packages/cozy-scripts/stack/quitStack.sh",
]
`;
